R/plugin.R

```diff
@@ -0,0 +1,130 @@
+orderly_db_plugin <- function() {
+  schema <- system.file("orderly.db.json", package = "orderly2.db",
+                        mustWork = TRUE)
+  orderly2::orderly_plugin(orderly_db_config,
+                           orderly_db_read,
+                           orderly_db_run,
+                           schema)
+}
+
+
+## Reads orderly_config.yml, configs plugin configuration. Evaluated
+## from the root directory.
+orderly_db_config <- function(data, filename) {
+  assert_named(data, unique = TRUE, name = sprintf("%s:orderly2.db", filename))
+  for (nm in names(data)) {
+    db <- data[[nm]]
+    prefix <- sprintf("%s:orderly2.db:%s", filename, nm)
+    check_fields(db, prefix, c("driver", "args"), NULL)
+    driver <- check_symbol_from_str(db$driver, paste0(prefix, ":driver"))
+    data[[nm]]$driver <- driver
+
+    assert_named(db$args, TRUE, paste0(prefix, ":args"))
+    ## TODO: instances here, once we support them, both instances and
+    ## default_instance field
+
+    ## There are two things to check with SQLite - first that we don't
+    ## use in-memory databases as we'll never reconnect to them. This
+    ## is probably not that useful a check but orderly does do it so
+    ## we may have added it for some reason.
+    ##
+    ## The other is more important though, which is is that database
+    ## files *should* be presented as relative paths, but these paths
```

Comment:
why "should"? what's wrong with absolute paths?

Reply:
if it's an absolute path it's unlikely to work on two different people's machines